src/operations/command.ts

```diff
@@ -107,13 +107,68 @@ export abstract class CommandOperation<T> extends AbstractCallbackOperation<T> {
     return true;
   }
 
-  executeCommand(
+  async executeCommand(
+    server: Server,
+    session: ClientSession | undefined,
+    cmd: Document
+  ): Promise<any> {
+    // TODO: consider making this a non-enumerable property
+    this.server = server;
+
+    const options = {
+      ...this.options,
+      ...this.bsonOptions,
+      readPreference: this.readPreference,
+      session
+    };
+
+    const serverWireVersion = maxWireVersion(server);
+    const inTransaction = this.session && this.session.inTransaction();
+
+    if (this.readConcern && commandSupportsReadConcern(cmd) && !inTransaction) {
+      Object.assign(cmd, { readConcern: this.readConcern });
+    }
+
+    if (this.trySecondaryWrite && serverWireVersion < MIN_SECONDARY_WRITE_WIRE_VERSION) {
+      options.omitReadPreference = true;
+    }
+
+    if (this.writeConcern && this.hasAspect(Aspect.WRITE_OPERATION) && !inTransaction) {
+      Object.assign(cmd, { writeConcern: this.writeConcern });
+    }
+
+    if (
+      options.collation &&
+      typeof options.collation === 'object' &&
+      !this.hasAspect(Aspect.SKIP_COLLATION)
+    ) {
+      Object.assign(cmd, { collation: options.collation });
+    }
+
+    if (typeof options.maxTimeMS === 'number') {
+      cmd.maxTimeMS = options.maxTimeMS;
+    }
+
+    if (this.hasAspect(Aspect.EXPLAINABLE) && this.explain) {
+      cmd = decorateWithExplain(cmd, this.explain);
+    }
+
+    return server.commandAsync(this.ns, cmd, options);
+  }
+}
+
+/** @internal */
+export abstract class CommandCallbackOperation<T = any> extends CommandOperation<T> {
+  constructor(parent?: OperationParent, options?: CommandOperationOptions) {
+    super(parent, options);
+  }
+
+  protected executeCommandCallback(
     server: Server,
     session: ClientSession | undefined,
     cmd: Document,
     callback: Callback
   ): void {
-    // TODO: consider making this a non-enumerable property
     this.server = server;
```

Comment:
Instead of having dupicate implementations, let's have the callback version invoke the async one.